kedro/framework/project/__init__.py

```diff
@@ -283,7 +315,28 @@ def find_pipelines() -> Dict[str, Pipeline]:
             function, the ``create_pipeline`` function does not return a
             ``Pipeline`` object, or if the module import fails up front.
     """
-    pipelines_dict = {"__default__": pipeline([])}
+    pipeline_module = None
+
+    # Handle the simplified project structure found in several starters.
+    pipeline_module_name = f"{PACKAGE_NAME}.pipeline"
+    try:
+        pipeline_module = importlib.import_module(pipeline_module_name)
+    except ModuleNotFoundError as exc:
+        if str(exc) != f"No module named '{pipeline_module_name}'":
+            warnings.warn(
+                IMPORT_ERROR_MESSAGE.format(
+                    module=pipeline_module_name, tb_exc=traceback.format_exc()
+                )
+            )
+    except:  # pylint: disable=bare-except  # noqa: E722
+        warnings.warn(
+            IMPORT_ERROR_MESSAGE.format(
+                module=pipeline_module_name, tb_exc=traceback.format_exc()
+            )
+        )
+
+    pipelines_dict = {"__default__": _create_pipeline(pipeline_module) or pipeline([])}
```

Comment:
Very minor, but it might be nice to make this a bit more consistent with the `pipelines` directory logic on L361 etc.

```suggestion
    except:  # pylint: disable=bare-except  # noqa: E722
        warnings.warn(
            IMPORT_ERROR_MESSAGE.format(
                module=pipeline_module_name, tb_exc=traceback.format_exc()
            )
        )
    else:
         pipeline_obj = _create_pipeline(pipeline_module)
    
    pipelines_dict = {"__default__": pipeline_obj or pipeline([])}
```
and set `pipeline_obj = None` on L318 instead of `pipeline_module`.

Reply:
Nice suggestion; done!